src/plots/barPlot.ts

```diff
@@ -439,10 +439,13 @@ export module Plots {
 
       let scale = <QuantitativeScale<any>>accScaleBinding.scale;
 
-      extents = extents.map((extent) => [
+      // To account for inverted domains
+      extents = extents.map((extent) => d3.extent([
         scale.invert(scale.scale(extent[0]) - this._barPixelWidth / 2),
-        scale.invert(scale.scale(extent[1]) + this._barPixelWidth / 2),
-      ]);
+        scale.invert(scale.scale(extent[0]) + this._barPixelWidth / 2),
+        scale.invert(scale.scale(extent[1]) - this._barPixelWidth / 2),
+        scale.invert(scale.scale(extent[1]) + this._barPixelWidth / 2)
```

Comment:
> 1. Extent is what you would get by autodomaining on any other type of plot.

Similar enough I think.  A plot gathers information about extents just in general, but this is most useful for autodomaining, since autodomaining will look at extents to determine the final domain.

> 2. It used to worked/works on vertical plots because extent[0] is always <= extent[1] --(assumption), so you could subtract 1/2 * barPixelWidth from extent[0], and add it to extent[1], and get a strictly wider range

I think the extents themselves were not really the problem but the scaled version of the extents.  For example, let's take Andrei's Before JSFiddle.  The yAxis domain is increasing starting from the bottom.  This means the scaled version of extent[0] is actually greater in value than the scaled version of extent[1] (since y counts increasing downwards)

> 3. So on horizontal, can extent[1] be less than extent[0]? So you account for that by always subtracting and adding from both extent[0] and extent[1]?

This was not necessarily the problem.  It's mainly that the assumptions about the scale were incorrect.